### ISSUE TITLE
Integrate release announcement

### ISSUE THREAD

**Issue by daniellekirkwood** [vector-im/element-web#25888]
As mentioned in an internal room, we've discussed having a 'release announcement'...

<table>
<tr>
 <td> Option
 <td> Pro's
 <td> Con's
<tr>
 <td> Full screen takeover, right hand side: <img width="500" alt="Screenshot 2023-07-31 at 15 04 34" src="https://github.com/vector-im/element-web/assets/89144281/4111395d-ebe7-4684-aef3-0c8952aee8f5">
<td> 

- Space for more content
- Must be dismissed/hard to miss

 <td>

- New build that's not aligned with current company strategy
- 

<tr>
 <td>
Top-left toasts:
<img width="267" alt="Screenshot 2023-07-31 at 15 08 45" src="https://github.com/vector-im/element-web/assets/89144281/c0de3c60-93d7-4080-9696-c955c76e8740">

 <td>

- Already exists
- Copy kept to a minimum (less to translate) and can link to a blog post

 <td>

- Easier to miss

<tr>
 <td> Specific tool tips: 
<img width="664" alt="Screenshot 2023-07-31 at 15 10 48" src="https://github.com/vector-im/element-web/assets/89144281/ad7a37d1-ed9b-4744-b78b-30fad3342c8f">
 <td> 

- Specific content will make the new UI easier to learn

 <td> 

- specific for each release (hard to repeat)
- New build that's not aligned with current company strategy

</table>

## Open questions
- Where does the data come from?

**Comment by nadonomy** [comment]
@germain-gg in any solution, if we wanted to only show it once to existing users, do we have an easy way to detect that?

**Comment by daniellekirkwood** [comment]
As discussed yesterday we're going to choose a simple option to unblock the first release planned. @germain-gg could you provide an estimate for the chalkboard experience (without thinking too much about the long-term issues we've raised). If we have one for the header and one for the right panel - is that a few hours/days/weeks?

**Comment by grati-cule** [comment]
Have we considered creating a room called something like Element Update Announcements that appears in the left hand panel and contains each announcement we send out? 

Element users are conditioned to consume messages this way so it seems intuitive to use it as the mechanism for announcements.

**Comment by HarHarLinks** [comment]
like https://matrix.to/#/#homeowners:matrix.org does for synapse? has the downside that it's impossible to show a diff from the currently installed version to the update.
or do you mean like a pseudo-room? maybe, but might as well be confusing e.g. if users expect to find that room in their other clients

**Comment by t3chguy** [comment]
@HarHarLinks it has an even larger downside of enumerating a lot of system admins into one list

**Comment by daniellekirkwood** [comment]
@germain-gg ping reminder :)

**Comment by americanrefugee** [comment]
@germain-gg @daniellekirkwood Here is the component for [Release Announcement](https://www.figma.com/file/rTaQE2nIUSLav4Tg3nozq7/Compound-Web-Components?type=design&node-id=1157-126112&mode=design&t=kVzK7rsCDEl7pLeq-4), which includes a proposal for the specific wording of each thing. WDYT?

![Release announcement](https://github.com/vector-im/element-web/assets/112560169/1873d86d-3a84-4569-a857-4f2c028ccc5c)

**Comment by daniellekirkwood** [comment]
Great. Couple o' questions...

- Do we have confidence that folks call it the "title bar"? I've only ever heard it be the "Room header"
- Will we be showing all of them at the same time, or one after another (if so, do they need numbers and what order are they in?) 
- I don't think we're changing the composer or the timeline for this first launch so we can keep those up our sleeve for a later date, right @germain-gg ?

**Comment by Johennes** [comment]
For https://github.com/vector-im/element-web/issues/25883, we'd only need the second and the third, yes. I think we should talk about how these tooltips carry over across versions though. Should they only show in the version the change was released in? Or should they still be displayed if somebody updates from an older version, skipping over the version where the change was released?

**Comment by daniellekirkwood** [comment]
> Should they only show in the version the change was released in? Or should they still be displayed if somebody updates from an older version, skipping over the version where the change was released?

I'm right in saying that we don't have a way to know if someone is new to the product or not?

I think it makes sense to show the tooltip _in the release_ and maybe the one after? But, as you've mentioned, if someone jumps a few releases they will miss the tooltips. 

Maybe that's the risk we have to take without an in-product release notes mechanism (which is a solution I would prefer). For example, Slack have a present icon so you can see what's new, and it's not intrusive. Beeper use a Matrix room to do so. @nadonomy have we explored these ideas and access points?

To unblock us for now, what about a time based approach? We're only going to be talking about the title bar and the right panel and that content is vanilla enough that if a new-to-element person sees it, it's not off putting. So maybe we show them for 4 weeks (see [data here](https://posthog.hss.element.io/insights/OlR24KGr)). It looks like after 2/3 weeks most people will have updated their version. 

We'd only want to show it to one person once though... that's doable, right?

**Comment by Johennes** [comment]
> I'm right in saying that we don't have a way to know if someone is new to the product or not?

We may not have that ability today but given that we can carry stored stated (e.g. your login) forward between updates, I think it should technically be possible for us to figure out if the user is on an initial installation or an update.

> Beeper use a Matrix room to do so

Some concerns about this option have been raised [here](https://github.com/vector-im/element-web/issues/25888#issuecomment-1674733036) and in the comment after.

> To unblock us for now, what about a time based approach? We're only going to be talking about the title bar and the right panel and that content is vanilla enough that if a new-to-element person sees it, it's not off putting. So maybe we show them for 4 weeks (see [data here](https://posthog.hss.element.io/insights/OlR24KGr)). It looks like after 2/3 weeks most people will have updated their version.

I like that. A time-based approach would also decouple the announcement logic from what our release cadence is entirely.

I'm actually wondering if we could even take it a step further and add _no_ expiration at all? We'd put the announcements into a file in the repository and they'd show on every update unless you have already seen them before. When we feel that the announcement isn't relevant anymore, we remove it from the file and it will stop showing for updaters with the next release.

> We'd only want to show it to one person once though... that's doable, right?

It should be, yes. We just need to store locally what announcements you have already seen.

**Comment by t3chguy** [comment]
> Beeper use a Matrix room to do so

This is false. They use something that looks like a Matrix room in the UI but is actually backed by an RSS feed.

**Comment by Johennes** [comment]
@daniellekirkwood given our capacity bottlenecks, I'm going to descope this from the current room header / details initiative because I'm fairly certain that we won't get to it in light of other work.

**Comment by daniellekirkwood** [comment]
@Johennes  I think we need _something_ - this is a big change otherwise. Are any of the options above a really quick, fast way of announcing changes?

**Comment by Johennes** [comment]
Option one and two from the issue description seem cheaper to implement than option three. However, option three seems like the most effective one. I agree that having this would be nice but that feels like a project of its own as we'd use such an infrastructure for other features, too. So I'm really not sure if this should block the epic when we're already struggling to find the time to finish it. 😕

**Comment by daniellekirkwood** [comment]
Ok, so then do we re-open this issue, remove it from the epic and attach it to a future compound epic when we _do_ have the time? This should stay on our backlog somewhere as we plan to do it in the future?

**Comment by Johennes** [comment]
Yes, that sounds like a good idea 👍
